### PR TITLE
Move to ubuntu 20.04/python 3.8 and add black/flake8 tests.

### PR DESCRIPTION
**Pull request by Roadmaster**
Main change is to migrate to ubuntu 20.04 and snap core20 as build/test/deploy platform.
Peripheral changes are adding flake and black and a makefile.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -12,11 +12,20 @@ on:
...
+        runs-on: ubuntu-20.04
         steps:
             - name: Checkout code
               uses: actions/checkout@v2
-            - name: Run tests
+            - name: Run unit tests
               run: |
-                  python -m unittest tests
-
+                 make
```
Comment (maxiberta) [inline .github/workflows/tests.yml:23]: This single `make` will run the first Makefile target, which is `test`, which is redundant, iiuc?

**Review by maxiberta — APPROVED** [review]
LGTM! +1 (with a tiny question)

File: .github/workflows/tests.yml (hunk 1 above)
Comment (Roadmaster) [reply .github/workflows/tests.yml:23]: You're right, I thought I'd removed this. Will update.

**Review by maxiberta — APPROVED** [review]
Thanks!

**Pull request merged**